openpype/hooks/pre_copy_last_published_workfile.py

```diff
@@ -109,11 +114,14 @@ def execute(self):
                 for subset in get_subsets(
                     project_name,
                     asset_ids=[asset_doc["_id"]],
-                    fields=["_id", "data.family", "data.families"],
+                    fields=["_id", "name", "data.family", "data.families"],
                 )
-                if subset["data"].get("family") == "workfile"
-                # Legacy compatibility
-                or "workfile" in subset["data"].get("families", {})
+                if (
+                    subset["data"].get("family") == "workfile"
+                    # Legacy compatibility
+                    or "workfile" in subset["data"].get("families", {})
+                )
+                and task_name in subset["name"]
```

Comment:
This condition is not right. You can "not use" task name in subset name, also in most of cases the task name have capitalized first letter if is used in subset name (so "animation" would become "Animation"). I think this requiers more complex logic that cannot be solved in single if statement inside loop...

Reply:
Something like this
```
        filtered_subsets = [
            subset
            for subset in get_subsets(
                project_name,
                asset_ids=[asset_doc["_id"]],
                fields=["_id", "name", "data.family", "data.families"],
            )
            if (
                subset["data"].get("family") == "workfile"
                # Legacy compatibility
                or "workfile" in subset["data"].get("families", {})
            )
        ]
        if not filtered_subsets:
            return

        matching_subset = None
        if len(filtered_subsets) > 1:
            low_task_name = task_name.lower()
            for subset in filtered_subsets:
                subset_name = subset["name"].lower()
                if low_task_name in subset_name:
                    matching_subset = subset
                    break

        if matching_subset is None:
            matching_subset = filtered_subsets[0]

        subset_id = matching_subset["_id"]
```

Reply:
Changed